install/installer/pkg/config/validation.go

```diff
@@ -55,6 +55,8 @@ func Validate(version ConfigVersion, cfg interface{}) (r *ValidationResult, err
 					res.Fatal = append(res.Fatal, fmt.Sprintf("Field '%s' is %s '%s'", v.Namespace(), tag, v.Param()))
 				case "startswith":
 					res.Fatal = append(res.Fatal, fmt.Sprintf("Field '%s' must start with '%s'", v.Namespace(), v.Param()))
+				case "block_new_users_passlist":
+					res.Fatal = append(res.Fatal, fmt.Sprintf("Field '%s' failed. If 'Enabled = true', there must be at least one fully-qualified domain name in the passlist", v.Namespace()))
```

Comment:
Maybe `Warning` is better, for the existing self-hosted

Reply:
No, I think fatal is correct. If you don't have any domains in the passlist then you end up in a broken state where the initial admin user gets blocked and you've basically bricked the installation. The only way I've found to resolve this is by deleting the database (I guess it's possible to do by deleting specific DB records, but I don't know those)

Reply:
It's good for new SH installation. But for existing SH, they need to LeanMore and change their `config.yaml`

And it blocks my case -- an empty passlist.

(My case also not allow me to use the same domain of email, so it's useless for my case like small company?)

Reply:
We can fix it by making first account available?

Reply:
Yeah, if there's an exception made to never block the first account then that would remove this problem completely.

I'd propose keeping the test as a fatal error, but it would remove the requirement for option `2` in the original test scenarios:
> If `blockNewUsers.enabled` is set to `true` and `blockNewUser.passlist` is set to `[]` - this should fail